src/ui/header/Header.tsx

```diff
@@ -5,10 +5,17 @@ import { Button } from '../button/Button';
 import { AppLocale } from 'context/locale/appLocale.enum';
 import { LocaleButton } from './localeButton/LocaleButton';
 import { useLocale } from 'hooks';
+import { HeaderProps } from './Header.types';
+import { useEffect, useState } from 'react';
 
-export const Header = () => {
+export const Header = ({ inputValue, onSubmit }: HeaderProps) => {
   const appLocaleValues = Object.values<AppLocale>(AppLocale);
   const { formatMessage, setLocale, locale } = useLocale();
+  const [inputValueState, setInputValueState] = useState(inputValue);
+
+  useEffect(() => {
+    setInputValueState(inputValue);
+  }, [inputValue]);
```

Comment:
Czy ten useEffect jest potrzebny? Bo jak rozumiem `inputValue` jest wartością tylko inicjalizacyjną, która jest przekazywana już w 14 linii

Reply:
Ten `useEffect` jest odpowiedzialny za aktualizację wartości inputa, kiedy usunie się filtr wyszukiwania kliknięciem w etykietę z x. Dopisałem `if (initialSearch !== inputValueState)`, żeby nie potrzebnie nie aktualizował tego stanu zaraz po jego zainicjowaniu.

Reply:
Mhm, ma to sens 👍